pkg/localstore/mode_put_test.go

```diff
@@ -484,87 +485,114 @@ func TestModePut_sameChunk(t *testing.T) {
-			t.Run("chunk on same index - timestamps in order", func(t *testing.T) {
-				db := newTestDB(t, nil)
-				// call unreserve on the batch with radius 0 so that
-				// localstore is aware of the batch and the chunk can
-				// be inserted into the database
-				first := generateTestRandomChunk()
-				second := generateTestRandomChunk()
-				stamp := first.Stamp()
-				ts := binary.BigEndian.Uint64(stamp.Timestamp())
-				buf := make([]byte, 8)
-				binary.BigEndian.PutUint64(buf, ts+1)
-				second = second.WithStamp(postage.NewStamp(stamp.BatchID(), stamp.Index(), buf, stamp.Sig()))
-				unreserveChunkBatch(t, db, 0, first, second)
-
-				_, err := db.Put(ctx, mode1, first)
-				if err != nil {
-					t.Fatal(err)
-				}
-				_, err = db.Put(ctx, mode2, second)
-				if err != nil {
-					t.Fatal(err)
-				}
-				newItemsCountTest(db.retrievalDataIndex, 1)(t)
-				newItemsCountTest(db.postageChunksIndex, 1)(t)
-				newItemsCountTest(db.postageRadiusIndex, 1)(t)
-				newItemsCountTest(db.postageIndexIndex, 1)(t)
-				newItemsCountTest(db.pullIndex, 1)(t)
-				_, err = db.Get(ctx, storage.ModeGetLookup, second.Address())
-				if err != nil {
-					t.Fatalf("expected no error, got %v", err)
-				}
-				_, err = db.Get(ctx, storage.ModeGetLookup, first.Address())
-				if !errors.Is(storage.ErrNotFound, err) {
-					t.Fatalf("expected %v, got %v", storage.ErrNotFound, err)
-				}
-			})
-			t.Run("chunk on same index - timestamps in reverse order", func(t *testing.T) {
-				db := newTestDB(t, nil)
-				// call unreserve on the batch with radius 0 so that
-				// localstore is aware of the batch and the chunk can
-				// be inserted into the database
-				first := generateTestRandomChunk()
-				second := generateTestRandomChunk()
-				stamp := first.Stamp()
-				ts := binary.BigEndian.Uint64(stamp.Timestamp())
-				buf := make([]byte, 8)
-				binary.BigEndian.PutUint64(buf, ts-1)
-				second = second.WithStamp(postage.NewStamp(stamp.BatchID(), stamp.Index(), buf, stamp.Sig()))
-				unreserveChunkBatch(t, db, 0, first, second)
-
-				_, err := db.Put(ctx, mode1, first)
-				if err != nil {
-					t.Fatal(err)
-				}
-				_, err = db.Put(ctx, mode2, second)
-				if err != nil {
-					t.Fatal(err)
-				}
-				newItemsCountTest(db.retrievalDataIndex, 1)(t)
-				newItemsCountTest(db.postageChunksIndex, 1)(t)
-				newItemsCountTest(db.postageRadiusIndex, 1)(t)
-				newItemsCountTest(db.postageIndexIndex, 1)(t)
-				newItemsCountTest(db.pullIndex, 1)(t)
-				_, err = db.Get(ctx, storage.ModeGetLookup, first.Address())
-				if err != nil {
-					t.Fatalf("expected no error, got %v", err)
-				}
-				_, err = db.Get(ctx, storage.ModeGetLookup, second.Address())
-				if !errors.Is(storage.ErrNotFound, err) {
-					t.Fatalf("expected %v, got %v", storage.ErrNotFound, err)
-				}
-			})
-		}
+	stamp := postagetesting.MustNewStamp()
+	ts := time.Now().Unix()
+
+	for _, tcn := range []struct {
+		name         string
+		mode         storage.ModePut
```

Comment:
if test cases are the same for all modes, then `mode` can be factored out to an enclosing `for _,mode := range []storage.ModePut{storage.ModePutSync,storage.ModePutRequest.....} { for ..... }`